src/tty/windows/mod.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright 2016 Joe Wilm, The Alacritty Project Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{self, Read, Write};
+use std::os::raw::c_void;
+use std::sync::Arc;
+
+use mio::{self, Evented, Poll, PollOpt, Ready, Token};
+use mio_anonymous_pipes::{EventedAnonRead, EventedAnonWrite};
+use mio_named_pipes::NamedPipe;
+
+use winapi::um::synchapi::WaitForSingleObject;
+use winapi::um::winbase::WAIT_OBJECT_0;
+use winapi::shared::winerror::WAIT_TIMEOUT;
+
+use config::Config;
+use display::OnResize;
+use cli::Options;
+use tty::EventedReadWrite;
+use term::SizeInfo;
+
+/// Handle to the winpty agent or conpty process. Required so we know when it closes.
+static mut HANDLE: *mut c_void = 0usize as *mut c_void;
+
+pub fn process_should_exit() -> bool {
+    unsafe {
+        match WaitForSingleObject(HANDLE, 0) {
+            // Process has exited
+            WAIT_OBJECT_0 => {
+                info!("wait_object_0");
+                true
+            }
+            // Reached timeout of 0, process has not exited
+            WAIT_TIMEOUT => false,
+            // Error checking process, winpty gave us a bad agent handle?
+            _ => {
+                info!("Bad exit: {}", ::std::io::Error::last_os_error());
+                true
+            }
+        }
+    }
+}
+
+mod conpty;
+mod winpty;
+
+pub enum PtyBackend<'a> {
+    Winpty(winpty::Agent<'a>),
+    Conpty(conpty::Conpty)
+}
+
+pub struct Pty<'a> {
+    // This Arc exists to allow threads other than the thread owning the Pty
+    // to safely resize the pty.
+    agent: Arc<PtyBackend<'a>>,
+    // TODO: It's on the roadmap for the Conpty API to support Overlapped I/O.
+    // See https://github.com/Microsoft/console/issues/262
+    // When support for that lands then it should be possible to use
+    // NamedPipe for the conout and conin handles rather than dynamic dispatch
```

Comment:
Should be tracked in github with an issue, that's likely a more visible solution than the code itself.